datajoint/table.py

```diff
@@ -72,7 +72,145 @@ def declare(self, context=None):
                 raise
         else:
             self._log('Declared ' + self.full_table_name)
+    
+    def preview_alter(self,new_definition):
+        """
+        Returns changes required to go from current defition to new defition.
+        Does not execute any changes. Use alter() method to finalize any changes.
+        Only supports table comment, and SECONDARY attribute changes(ADD,DROP,CHANGE).
+        Reordering currently not supported.
+        Use # {<old_name>} <comment> to rename given attribute.(insert values between '< >') 
+
+        :param new_definition: string, same format used to create the table.
+        :return: alter sql. 
+        """
+
+        alter_sql = '' #alter sql
+
+        #change in table comment
+        new_definition = re.split(r'\s*\n\s*', new_definition.strip())
+        new_table_comment = new_definition.pop(0)[1:].strip() if new_definition[0].startswith('#') else ''
+        
+        if (self.heading.table_info['comment'] != new_table_comment):    
+            alter_sql += ('COMMENT = "' + new_table_comment + '", ')
+        
+        new_attributes = []
+        in_key = True
+        
+        for line in new_definition:
+            if line.startswith('#'):
+                continue
+            elif line.startswith('---') or line.startswith('___'):
+                in_key = False
+            elif is_foreign_key(line):
+                continue
+            elif re.match(r'^(unique\s+)?index[^:]*$', line, re.I): # index
+                continue
+            else:
+                if not in_key:
+                    # change in secondary attributes
+                    try:
+                        attr = attribute_parser.parseString(line+'#', parseAll=True)
+                    except pp.ParseException as err:
+                        raise DataJointError('Declaration error in position {pos} in line:\n  {line}\n{msg}'.format(
+                            line=err.args[0], pos=err.args[1], msg=err.args[2]))
+                    
+                    #external not supported
+                    if attr['type'].startswith('external'):
+                        continue
+
+                    #old_name and name are the same if no rename
+                    attr['old_name'] = attr['name']
+                    attr['comment'] = attr['comment'].rstrip('#')
+                    
+                    #extract rename, and new comment if necessary. brackets after any alphanumerical are ignored.
+                    rename = re.match(r"[^\w]*{[a-z][a-z0-9_]*}", attr['comment'])
```

Comment:
Try
```suggestion
                    rename = re.match(r'({(?P<name>\w+)\s*})?\s*(?P<comment>.*)', attr['comment'])
                    if rename.group('name'):
                        ...
                    
```

Reply:
`r'{\s*(?P<name>\w+)\s*}\s*(?P<comment>.*)'`
legal: 234_f. allows for illegal attribute names like digit only

`r'{\s*(?P<name>[a-z][a-z0-9_]*)\s*}\s*(?P<comment>.*)'`
How about this: alphanumerical lowercase must start with a letter
fails: # {er} sf. space after # in attribute comment

this?: added \s* in the front to deal with blank space before '{'
`r'\s*{\s*(?P<name>[a-z][a-z0-9_]*)\s*}\s*(?P<comment>.*)'`

Reply:
Oh, the main thing I was pointing out is that you can use `rename.group('name')` inside the condition rather than parsing again.   So yes, please use the more precise regular expression.

Reply:
https://github.com/datajoint/datajoint-python/blob/c12dd9a936553bc52610b7193ab26208778f1bbf/datajoint/table.py#L129-L132